cli/dcos-spark/cli_test.go

```diff
@@ -177,3 +177,49 @@ func TestPayloadWithSecret(t *testing.T) {
 	checkSecret(keytab, keytab, t)
 	checkSecret(keytab_prefixed, keytab, t)
 }
+
+func TestSaslSecret(t *testing.T) {
+	inputArgs := fmt.Sprintf(
+		"--executor-auth-secret /%s " +
+			"--class %s "+
+			"%s --input1 value1 --input2 value2", sparkAuthSecret, mainClass, appJar)
+
+	cmd := SparkCommand{
+		"subId",
+		inputArgs,
+		image,
+		space,
+		make(map[string]string),
+		"",
+		false,
+		false,
+		0,
+		"",
+	}
+	payload, err := buildSubmitJson(&cmd)
+
+	m := make(map[string]interface{})
+
+	json.Unmarshal([]byte(payload), &m)
+
+	if err != nil {
+		t.Errorf("%s", err.Error())
+	}
+
+	stringProps := map[string]string{
+		"spark.authenticate": "true",
+		"spark.mesos.containerizer": "mesos",
+		"spark.authenticate.enableSaslEncryption": "true",
+		"spark.authenticate.secret": "spark_shared_secret",
+		"spark.executorEnv._SPARK_AUTH_SECRET": "spark_shared_secret",
+		"spark.mesos.driver.secret.filenames": sparkAuthSecret,
+		"spark.mesos.driver.secret.names": fmt.Sprintf("/%s", sparkAuthSecret),
```

Comment:
@ArtRand What was the reason for adding a "/"? I remember it working fine without the slash?

Reply:
It will work either way, if there is a `/` when you specify `--executor-auth-secret /secret` then it  will be propagated. If the secret is at root, it's unnecessary.

Reply:
Ok